### PR TITLE
Add doc for dedicated cloud encryption

### PR DESCRIPTION
**Pull request by abhinavg6**
## Description

Add a doc for outlining the state of encryption on Dedicated Cloud, and the caveat for customers if they want to bring their own key.

## Ticket

NA

## Checklist

Check if your PR fulfills the following requirements. Put an `X` in the boxes that apply.

- [x] Files I edited were previewed on my local development server with `yarn start`. My changes did not break the local preview.
- [x] Build (`yarn docusaurus build`) was run locally and successfully without errors or warnings.
- [x] I merged the latest changes from `main` into my feature branch before submitting this PR.

### CODE REVIEW

File: docs/guides/hosting/data-security/data-encryption.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+---
+displayed_sidebar: default
+---
+
+# Encryption on Dedicated Cloud
+
+W&B uses a W&B-managed cloud-native key to encrypt the W&B-managed database and object storage in every [Dedicated Cloud](../hosting-options/dedicated_cloud.md), by using the customer-managed encryption key (CMEK) capability in each cloud. In this case, W&B acts as a `customer` of the cloud provider, while providing the W&B platform as a service to you. Using a W&B-managed key means that W&B has control over the keys that are used to encrypt the data in each cloud, thus doubling down on its promise to provide a highly safe and secure platform to all of its customers.
```
Comment (zacharyblasczyk) [inline docs/guides/hosting/data-security/data-encryption.md]: Looks good.

Comment (zacharyblasczyk) [inline docs/guides/hosting/data-security/data-encryption.md]: Looks good

File: docs/guides/hosting/data-security/data-encryption.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+If your organization needs to use their own cloud-native key to encrypt the W&B-managed database and object storage in order to approve the use of Dedicated Cloud for your AI workflows, W&B can review it on a exception basis. If approved, use of your cloud-native key for encryption will be bound to the `shared responsibility model` of W&B Dedicated Cloud. If your key is removed or if W&B's IAM access to that key is revoked at any point when your Dedicated Cloud instance is live, W&B will not be liable for any resulting data loss or corruption and will not be responsible for recovery of such data.
```
Comment (zacharyblasczyk) [inline docs/guides/hosting/data-security/data-encryption.md]: Really good phrasing, but we probably need a link.

**Review by zacharyblasczyk — APPROVED** [review]

File: docs/guides/hosting/data-security/data-encryption.md (hunk 2 above)
Comment (abhinavg6) [reply docs/guides/hosting/data-security/data-encryption.md]: Yeah, we're gonna ask the security team to add a shared responsibility doc at some point soon. Until then it'll be a temp placeholder :)

**Review by ngrayluna — APPROVED** [review]

**Pull request merged**